batchflow/models/tf/unet.py

```diff
@@ -87,3 +88,140 @@ def build_config(self, names=None):
+                    If int, same number of filters applies to all layers on all stages
+                    If list of ints, specifies number of filters in each layer of different stages
+                    If list of list of ints, specifies number of filters in different layers on different stages
+                    If not given or None, filters parameters in encoder/blocks, decoder/blocks and decoder/upsample
+                    default to values which make number of filters double
+                    on each stage of encoding and halve on each stage of decoding,
+                    provided that `decoder/skip` is `True`. Specify `filters=None` explicitly
+                    if you want to use custom `num_steps` and infer `filters`
+
+        decoder : dict
+            num_stages : int
+                number of upsampling blocks. Defaults to the number of downsamplings.
+
+            factor : None, int or list of ints
+                If int, the total upsampling factor for all stages combined.
+                If list, upsampling factors for each stage
+                If not given or None, defaults to [2]*num_stages
+
+            blocks : dict
+                Parameters for post-processing blocks:
+
+                filters : None, int, list of ints or list of lists of ints
+                    same as encoder/blocks/filters
+
+            upsample : dict
+                Parameters for upsampling (see :func:`~.layers.upsample`).
+
+                filters : int, list of ints or list of lists of ints
+                    same as encoder/blocks/filters
+
+            dense : dict
+                Parameters for dense block on skip connections including parameters for conv_block
+
+                upsample : dict
+                    upsample parameters for block
+
+    for more parameters see (see :class:`~.EncoderDecoder`)
+    """
+    @classmethod
+    def default_config(cls):
+        config = super().default_config()
+        return config
+
+    def build_config(self, names=None):
+        config = super().build_config(names)
+
+        num_stages = config.get('body/encoder/num_stages')
+
+        if config.get('body/decoder/dense/upsample') is None:
+            config['body/decoder/dense/upsample'] = dict(layout='tna')
+
+        if config.get('body/decoder/dense/filters') is None:
+            config['body/decoder/dense/filters'] = [32 * 2**i for i in range(num_stages)]
+
+        if config.get('body/decoder/dense/layout') is None:
+            config['body/decoder/dense/layout'] = 'cna'
+
+        if config.get('body/decoder/dense/kernel') is None:
+            config['body/decoder/dense/kernel'] = 3
+
+        if config.get('body/decoder/dense/combine_op') is None:
+            config['body/decoder/dense/combine_op'] = 'concat'
+
+        return config
+
+    @classmethod
+    def decoder(cls, inputs, name='decoder', **kwargs):
+        decoder_filters = kwargs['upsample']['filters']
```

Comment:
```suggestion
        decoder_filters = kwargs['upsample/filters']
```
can improve readability everywhere

Reply:
`kwargs` is ordinal dict